mozilla/fx-private-relay
# Domain registration confirmation modal

This PR adds the Domain Registration modal. 

To test: 

- Create/log into PREMIUM account who hasn't yet registered a subdomain
- Fill out form, click "register domain" button
- **Expected:**
  - If the domain is available: The user should see a modal asking them to confirm their selection. 
  - If the domain is not available (because it's either already registered or a disallowed word), the form should NOT show the modal and should submit as normal. This will cause the page to reload with a new error message displayed. 

Note: The confirmation modal submit button should be disabled until the user clicks the checkbox. 

Other UX Notes: 
- Clicking anywhere outside of the modal should dismiss it
- Clicking the ESC key should dismiss the modal 
- Clicking the modal cancel button should dismiss the modal

Completes MPP-406
Related to #1043

Screenshot: https://drive.google.com/file/d/1EcPJBC27kR1bG1m8E_sfKME0BjUoldUV/view?usp=sharing
